### PR TITLE
Adding louvain tarballs

### PR DESCRIPTION
**Pull request by KrainskiL**

### CODE REVIEW

File: L/louvain/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    GitSource("https://github.com/KrainskiL/louvain.git", "9e3e081f054e531e32af38fe71d2991f954ea347")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd louvain/
+make
+mv hierarchy louvain matrix convert $prefix
+exit
```
Comment (giordano) [inline L/louvain/build_tarballs.jl:19]:
```suggestion
cd $WORKSPACE/srcdir/louvain/
make -j${nproc}
mv hierarchy${exeext} louvain${exeext} matrix${exeext} convert${exeext} ${bindir}/.
```

File: L/louvain/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("hierarchy", :hierarchy),
+    ExecutableProduct("louvain", :louvain),
+    ExecutableProduct("convert", :convertlouvain)
+]
```
Comment (giordano) [inline L/louvain/build_tarballs.jl:31]: Missing `matrix`?

Comment (KrainskiL) [reply L/louvain/build_tarballs.jl:29]: `matrix` executable is not needed in current release, but may be needed in the future. Still it stays in build script for convenience. Thank you for suggestions!

File: L/louvain/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/KrainskiL/louvain.git", "9e3e081f054e531e32af38fe71d2991f954ea347")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/louvain/
+make -j${nproc}
+mv hierarchy${exeext} louvain${exeext} matrix${exeext} convert${exeext} ${bindir}/.
```
Comment (giordano) [inline L/louvain/build_tarballs.jl:17]: Ok, let's do this:
```suggestion
mkdir -p ${bindir}
for bin in hierarchy louvain matrix convert; do
    mv "${bin}" "${bindir}/${bin}${exeext}"
done
```

File: L/louvain/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/louvain/
+make -j${nproc}
+mkdir -p ${bindir}
+for exe in hierarchy louvain matrix convert; do
+    mv "${exe}" "${bindir}/${exe}${exeext}"
+done
+"""
```
Comment (giordano) [inline L/louvain/build_tarballs.jl:21]: We should also install the license files:
```suggestion
install_license *gpl*.txt
"""
```
:wink:

File: L/louvain/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline L/louvain/build_tarballs.jl:25]:
```
┌ Warning: /agent/_work/1/s/L/louvain/build/i686-linux-gnu/X8bvrzOi/destdir/bin/louvain contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/TCJai/src/auditor/compiler_abi.jl:222
```
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: L/louvain/build_tarballs.jl (hunk 4 above)
Comment (KrainskiL) [reply L/louvain/build_tarballs.jl:21]: Good point. Thank you.

**Comment by giordano** [comment]
Looks all good now, thanks for the contribution!

**Pull request merged**